packages/cozy-client/src/CozyClient.js

```diff
@@ -804,21 +804,28 @@ client.query(Q('io.cozy.bills'))`)
   async query(queryDefinition, { update, ...options } = {}) {
     this.ensureStore()
     const queryId = options.as || this.generateId(queryDefinition)
-    this.ensureQueryExists(queryId, queryDefinition)
-
-    if (options.fetchPolicy) {
-      if (!options.as) {
-        throw new Error(
-          'Cannot use `fetchPolicy` without naming the query, please use `as` to name the query'
-        )
-      }
-
-      const existingQuery = this.getQueryFromState(queryId)
-      const shouldFetch = options.fetchPolicy(existingQuery)
-      if (!shouldFetch) {
-        return
+    const existingQuery = this.getQueryFromState(queryId)
+    // If we already have the same query in loading state, no
+    // need to refrech it
+    if (existingQuery.fetchStatus === 'loading') {
+      return
+    }
+    if (existingQuery) {
```

Comment:
This check should be done before the `fetchStatus` one to avoid accessing an attribute of `undefined`.